core/workspace.ts

```diff
@@ -634,12 +634,12 @@ export class Workspace implements IASTNodeLocation {
       if (!event) continue;
       events.push(event);
     }
+    events = eventUtils.filter(events, redo);
```

Comment:
Is this change still necessary now that we are filtering the undo stack itself?

Reply:
It may still be semantically correct to filter events being processed in the undo stack before pushing them to the redo stack. Otherwise, Blockly ends up dropping events from the queue that happen to still be on the stack, which ultimately leads to what happened in the initial issue since there is an event that never gets cleaned up from the stack.

Reply:
So is this just a precautionary thing? My reading is that they're getting filtered before being run + pushed to the undo stack anyway because they're being filtered before they're even pushed to the undo stack.

Reply:
Hmm. That is a fair point. I can push an update with only the second commit. If there continue to be occasional issues then we can always reapply it.